api/v1/helpers/verbs/doDelete.js

```diff
@@ -42,11 +43,22 @@ function doDelete(req, res, next, props) {
       )
       .then((o) => u.isWritable(req, o,
           featureToggles.isFeatureEnabled('enforceWritePermission')))
-      .then((o) => o.destroy());
+      .then((o) => {
+        /*
+         * An empty array is returned when destroy is called on an instance
+         * of a model that has hard delete turned on. We still want to return
+         * the deleted instance in all the cases. So, before the instance
+         * is destroyed it is saved and for models with hard deleted turned on
+         * this saved instance is returned
+         */
+        obj = o;
```

Comment:
I don't think you should add it here, but let's make sure the swagger definition of the response schema is right